src/shared/components/post/post-form.tsx

```diff
@@ -439,56 +415,121 @@ export class PostForm extends Component<PostFormProps, PostFormState> {
+                enableNsfw={this.props.enableNsfw}
+                allLanguages={this.props.allLanguages}
+                siteLanguages={this.props.siteLanguages}
+                viewOnly
+                // All of these are unused, since its view only
+                onPostVote={() => {}}
+                onPostReport={() => {}}
+                onBlockPerson={() => {}}
+                onLockPost={() => {}}
+                onDeletePost={() => {}}
+                onRemovePost={() => {}}
+                onSavePost={() => {}}
+                onFeaturePost={() => {}}
+                onPurgePerson={() => {}}
+                onPurgePost={() => {}}
+                onBanPersonFromCommunity={() => {}}
+                onBanPerson={() => {}}
+                onAddModToCommunity={() => {}}
+                onAddAdmin={() => {}}
+                onTransferCommunity={() => {}}
+              />
+            </>
+          )
+        );
+      }
+    }
+  }
 
+  async handlePostSubmit(i: PostForm, event: any) {
+    event.preventDefault();
     // Coerce empty url string to undefined
     if ((i.state.form.url ?? "blank") === "") {
       i.setState(s => ((s.form.url = undefined), s));
     }
+    i.setState({ loading: true });
+    const auth = myAuthRequired();
 
     let pForm = i.state.form;
     let pv = i.props.post_view;
-    let auth = myAuth();
-    if (auth) {
-      if (pv) {
-        let form: EditPost = {
-          name: pForm.name,
-          url: pForm.url,
-          body: pForm.body,
-          nsfw: pForm.nsfw,
-          post_id: pv.post.id,
-          language_id: pv.post.language_id,
-          auth,
-        };
-        WebSocketService.Instance.send(wsClient.editPost(form));
-      } else {
-        if (pForm.name && pForm.community_id) {
-          let form: CreatePost = {
-            name: pForm.name,
-            community_id: pForm.community_id,
-            url: pForm.url,
-            body: pForm.body,
-            nsfw: pForm.nsfw,
-            language_id: pForm.language_id,
-            honeypot: pForm.honeypot,
-            auth,
-          };
-          WebSocketService.Instance.send(wsClient.createPost(form));
-        }
-      }
+    if (pv) {
+      await i.props.onEdit?.({
+        name: pForm.name,
+        url: pForm.url,
+        body: pForm.body,
+        nsfw: pForm.nsfw,
+        post_id: pv.post.id,
+        language_id: pv.post.language_id,
+        auth,
+      });
+    } else if (pForm.name && pForm.community_id) {
+      await i.props.onCreate?.({
+        name: pForm.name,
+        community_id: pForm.community_id,
+        url: pForm.url,
+        body: pForm.body,
+        nsfw: pForm.nsfw,
+        language_id: pForm.language_id,
+        honeypot: pForm.honeypot,
+        auth,
+      });
     }
+
+    i.setState({ loading: false });
   }
 
-  copySuggestedTitle(i: PostForm) {
-    let sTitle = i.state.suggestedTitle;
+  copySuggestedTitle(d: { i: PostForm; suggestedTitle?: string }) {
```

Comment:
This could just receive the `suggestedTitle` as a prop, and set the form.title to that.